internal/services/compute/disk_encryption_set_resource.go

```diff
@@ -118,18 +128,47 @@ func resourceDiskEncryptionSetCreate(d *pluginsdk.ResourceData, meta interface{}
 	}
 
 	keyVaultKeyId := d.Get("key_vault_key_id").(string)
+
+	if keyVaultKeyId == "" {
+		return fmt.Errorf("missing required argument. The argument %q is required, but no definition was found", "key_vault_key_id")
+	}
+
 	keyVaultDetails, err := diskEncryptionSetRetrieveKeyVault(ctx, keyVaultsClient, resourcesClient, keyVaultKeyId)
 	if err != nil {
 		return fmt.Errorf("validating Key Vault Key %q for Disk Encryption Set: %+v", keyVaultKeyId, err)
 	}
 
 	if keyVaultDetails != nil {
 		if !keyVaultDetails.softDeleteEnabled {
-			return fmt.Errorf("validating Key Vault %q (Resource Group %q) for Disk Encryption Set: Soft Delete must be enabled but it isn't!", keyVaultDetails.keyVaultName, keyVaultDetails.resourceGroupName)
+			return fmt.Errorf("validating Key Vault %q (Resource Group %q) for Disk Encryption Set: Soft Delete must be enabled", keyVaultDetails.keyVaultName, keyVaultDetails.resourceGroupName)
 		}
 	}
 
 	rotationToLatestKeyVersionEnabled := d.Get("auto_key_rotation_enabled").(bool)
+
+	// NOTE: The API requires a versioned key to be sent however if rotationToLatestKeyVersion is enabled this will cause
+	// terraform to revert the rotated key to the previous version that is defined in the configuration file...
+	// Issue #22864
+	if rotationToLatestKeyVersionEnabled {
+		if err := diskEncryptionSetValidateRotationPolicy(rotationToLatestKeyVersionEnabled, keyVaultKeyId); err != nil {
+			return err
+		}
```

Comment:
With the changes noted above, we should be able to remove this?

Reply:
Fixed.